### PR TITLE
Replace container registery with artifact registry in CloudBuild

### PR DESCRIPTION
**Pull request by renato-rudnicki**
Fixes https://github.com/terraform-google-modules/terraform-example-foundation/issues/366

This PR replaces `gcr.io/$PROJECT_ID/terraform` with `$_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/prj-tf-runners/terraform` in the CloudBuild files.

Result with the fix:

```
Starting Step #0 - "setup"
Step #0 - "setup": Pulling image: us-east1-docker.pkg.dev/prj-cloudbuild-34b9/prj-tf-runners/terraform
Step #0 - "setup": Using default tag: latest
Step #0 - "setup": latest: Pulling from prj-cloudbuild-34b9/prj-tf-runners/terraform
Step #0 - "setup": 75f546e73d8b: Already exists
```

### CODE REVIEW

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the quick fix @renato-rudnicki 
Lets switch to using`_GAR_REPOSITORY` https://github.com/terraform-google-modules/terraform-google-bootstrap/pull/97 from `2.2.1` release

**Comment by daniel-cit** [comment]
Using version  [v2.2.1](https://github.com/terraform-google-modules/terraform-google-bootstrap/releases/tag/v2.2.1) of the `terraform-google-bootstrap` module to create the infra structure, the value set for the environment variable `_GAR_REPOSITORY`  in the cloud build triggers is `prj-tf-runners`.

Would the final form of the fix be something like this?

```
name: $_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/$_GAR_REPOSITORY/terraform
```

@bharathkkb

**Comment by bharathkkb** [comment]
@daniel-cit yeah that looks right. Users of example-foundation can also pass a custom repo via `gar_repo_name` and by using `$_GAR_REPOSITORY` we can get that in CB.

**Comment by daniel-cit** [comment]
With the last commit the docker build steps in the build yaml files are now using the `_GAR_REPOSITORY` variable:

```
steps:
- id: 'setup'
  name: $_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/$_GAR_REPOSITORY/terraform
  entrypoint: /bin/bash
  args:
```

The result with the new version:

```     
Starting Step #0 - "setup"
Step #0 - "setup": Pulling image: us-east1-docker.pkg.dev/prj-cloudbuild-9473/prj-tf-runners/terraform
Step #0 - "setup": Using default tag: latest
Step #0 - "setup": latest: Pulling from prj-cloudbuild-9473/prj-tf-runners/terraform
```

File: build/cloudbuild-tf-apply.yaml [hunk 1]
```diff
@@ -17,11 +17,11 @@ substitutions:
   _POLICY_REPO: '/workspace/policy-library' # add path to policies here https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#how-to-use-terraform-validator
 steps:
 - id: 'setup'
-  name: gcr.io/$PROJECT_ID/terraform
+  name: $_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/$_GAR_REPOSITORY/terraform
   entrypoint: /bin/bash
   args:
   - -c
-  - |
+  - |''
```
Comment (bharathkkb) [inline build/cloudbuild-tf-apply.yaml:24]: Is this needed?

**Review by bharathkkb — APPROVED** [review]
LGTM with one question

File: build/cloudbuild-tf-apply.yaml (hunk 1 above)
Comment (daniel-cit) [reply build/cloudbuild-tf-apply.yaml:24]: Good catch! 
This line should not have been changed. We will revert this it.

**Pull request merged**